thunder/torch/__init__.py

```diff
@@ -1503,6 +1503,39 @@ def exp2_(a):
     return prims.copy_(exp2(a), a)
 
 
+# fake out of place variant
+@torchsymbol(id="exponential")
+def exponential(a, rate=1, generator=None):
+    assert generator is None
```

Comment:
Instead of asserting, check that the generator is None. That will create a clearer error message.

Also, Python executed in optimized mode won't check asserts, which are intended for debugging and testing